elide-core/src/main/java/com/yahoo/elide/core/DataStoreTransaction.java

```diff
@@ -207,4 +221,17 @@ default void preCommit() {
 
         return val;
     }
+
+    default <T> Object getRelationWithSortingAndPagination(
```

Comment:
I think it would be nice not to require a separate method for handling sorting and pagination. I think this entered the code as more of a band-aid. If we're going to go through a major refactoring of this in the first place, then we should consider consolidating this functionality.

One effective way may be to always give sane defaults in the event sorting and/or pagination is not specified.